apps/emails-and-messages/src/modules/sendgrid/configuration/sendgrid-config.ts

```diff
@@ -1,26 +1,22 @@
+import { MessageEventTypes } from "../../event-handlers/message-event-types";
+
+export interface SendgridEventConfiguration {
+  active: boolean;
+  eventType: MessageEventTypes;
+  template: string;
+}
+
 export interface SendgridConfiguration {
+  id: string;
   active: boolean;
   configurationName: string;
   sandboxMode: boolean;
-  senderName: string;
-  senderEmail: string;
+  senderName?: string;
+  senderEmail?: string;
   apiKey: string;
-  templateInvoiceSentSubject: string;
-  templateInvoiceSentTemplate: string;
-  templateOrderCancelledSubject: string;
-  templateOrderCancelledTemplate: string;
-  templateOrderConfirmedSubject: string;
-  templateOrderConfirmedTemplate: string;
-  templateOrderFullyPaidSubject: string;
-  templateOrderFullyPaidTemplate: string;
-  templateOrderCreatedSubject: string;
-  templateOrderCreatedTemplate: string;
-  templateOrderFulfilledSubject: string;
-  templateOrderFulfilledTemplate: string;
+  events: SendgridEventConfiguration[];
```

Comment:
Eventually, I'll end up with such a setup. Right now the schemas differ a bit from the ts types in both providers (some of the fields are kept different because of the way the form is constructed).

For the sake of consistency, I'll keep it and refactor it during the creation of shared classes for providers to remove duplicated code.